### PR TITLE
Fixes to llvm backend for iele

### PR DESCRIPTION
**Pull request by dwightguth**
Includes: 
* support for matching and compacting overloaded productions
* update to latest version of Array in K
* make llvm backend compatible with building a library
* code to free kore memory
* bug fixes

### CODE REVIEW

File: runtime/take_steps.ll [hunk 1]
```diff
@@ -8,14 +8,20 @@ declare %block* @step(%block*)
 declare void @finish_rewriting(%block*) #0
 declare void @koreCollect(%block **)
```
Comment (pmeredit) [inline runtime/take_steps.ll]: Why are we writing these files directly in llvm instead of using C++ with C linkage?

Comment (dwightguth) [reply runtime/take_steps.ll]: Just the decision that was made at the time, no particular reason. We could convert them all at a later date, but that should be a separate PR.

**Comment by dwightguth** [comment]
Jenkins: test this please

File: lib/codegen/CreateTerm.cpp [hunk 2]
```diff
@@ -453,6 +453,7 @@ llvm::Value *CreateTerm::notInjectionCase(KOREObjectCompositePattern *constructo
   return new llvm::BitCastInst(Block, llvm::PointerType::getUnqual(Module->getTypeByName(BLOCK_STRUCT)), "", CurrentBlock);
 }
 
+// returns a value and a boolean indicating whetther that value could be an injection
```
Comment (pmeredit) [inline lib/codegen/CreateTerm.cpp]: whetther => whether

**Pull request merged**